### ISSUE TITLE
关于训练vgg16,resnet18时不收敛的问题

### ISSUE THREAD

**Issue by HJXzjut** [zhoumingyi/DaST#8]
你好，我自己写了一个基于Cifar10的生成器，然后用vgg16作为替代模型，被攻击模型选的也是vgg16，但是我在进行替代模型训练时似乎没有办法达到收敛，训练出来的模型在cifar10测试集上只有百分之十左右的精确度，而且使用目标攻击的成功率几乎也就是在百分之十左右。
我注意到论文中也有用cifar10数据集测试vgg,resnet等模型的实验结果，虽然文中的resnet目标攻击成功率也是在个位数，但是vgg能达到百分之二十几以上，因此我的实验结果显然是有问题的。想请教一下我的问题是不是出在了参数的设定上，还是说生成器的结构也会影响我的实验结果？
如果是参数的问题，想请教一下训练vgg替身模型（基于cifar10数据结构的生成器作为训练集）的一些参（lr,batch_num,epoch），开源代码只有mnist生成器以及相应的参数。万分感激！

**Comment by zhoumingyi** [comment]
等我先自己测试下，稍等

**Comment by zhoumingyi** [comment]
生成器的结构确实会影响结果，要具体讨论的话你方便发一下你的训练代码给我吗？

**Comment by HJXzjut** [comment]
万分感激您在百忙之中回信！附上我训练替身模型的代码（主要在您的源码中更改了模型以及导入模型的方法，还有生成器的结构，注释了一部分用于测试攻击成功率的代码）


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "zhoumingyi/DaST"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月27日(星期一) 下午5:53
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [zhoumingyi/DaST] 关于训练vgg16,resnet18时不收敛的问题 (#8)





 
生成器的结构确实会影响结果，要具体讨论的话你方便发一下你的训练代码给我吗？
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by HJXzjut** [comment]
您好，我就是在github上提问的，和刚刚发你代码的就是同一个人。刚刚我忘记附上我加载模型的函数了，抱歉&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "zhoumingyi/DaST"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月27日(星期一) 下午5:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [zhoumingyi/DaST] 关于训练vgg16,resnet18时不收敛的问题 (#8)





 
等我先自己测试下，稍等
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zhoumingyi** [comment]
> 万分感激您在百忙之中回信！附上我训练替身模型的代码（主要在您的源码中更改了模型以及导入模型的方法，还有生成器的结构，注释了一部分用于测试攻击成功率的代码）

我好像没收到你的代码，可以单独用邮件打包发送到mingyi.zhou@monash.edu吗？

**Comment by HJXzjut** [comment]
I'm sorry because I didn't succeed either, and since the author has been busy for a while and hasn't replied to me, maybe you should ask the author.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "zhoumingyi/DaST"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年7月7日(星期四) 晚上10:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [zhoumingyi/DaST] 关于训练vgg16,resnet18时不收敛的问题 (#8)





 
Is this issue resolved ? I am also trying to replicate this for cifar-10, but so far no success.

**Comment by zacharias-qi** [comment]
作者您好，我这边也是遇到了CIFAR-10上面测试结果不收敛的问题。我是使用VGG13作为substitute model来攻击预训练的VGG16（测试精度93.4%）。 具体参数我使用的是：G_type = 1, alpha = 0.2, beta = 0.1 和 1.0， 但是结果都不理想，在beta=1.0 (个人理解为DaST-P)， 我的替代模型精度和攻击成功率结果仅有10%左右。希望可以讨教一下具体在CIFAR-10上面应该如何设置实验，非常感谢！

**Comment by zhoumingyi** [comment]
@HJXzjut @shubham303 @zacharias-qi Sorry to reply so late because I need to handle many research and personal stuffs. I have uploaded the code for CIFAR-10. But you need to note the training on CIFAR-10 is very unstable. As the provided CIFAR-10 log file, I tried twice but the first time the model's training collapsed. The training on CIFAR-10 is something tricky, the two separate losses for the G should be balanced. So unlike the training on MNIST, I reduce the beta to 0.1 because sometimes the loss_imitate can not decrease, and the G can just minimize the loss_diversity. 

I have tried adding the random seed, but I don't know why it didn't work. So maybe you should try several times to train the substitute model. You can observe the losses. If the training is collapsed, you stop it early. The good training for DaST-P should be like the second example in the cifar10 log file. But for DaST-L, The training will always be collapsed after multiple epochs, you can observe it in my paper, the performance of DaST-L is not good for CIFAR-10.

**Comment by zhoumingyi** [comment]
BTW, in the CIFAR-10 code, I removed the AdverTorch library and used the Foolbox instead. Because the AdverTorch doesn't update for a long time, It has some problems when we use it under recent PyTorch environments.

**Comment by zacharias-qi** [comment]
Sincrely

> @HJXzjut @shubham303 @zacharias-qi Sorry to reply so late because I need to handle many research and personal stuffs. I have uploaded the code for CIFAR-10. But you need to note the training on CIFAR-10 is very unstable. As the provided CIFAR-10 log file, I tried twice but the first time the model's training collapsed. The training on CIFAR-10 is something tricky, the two separate losses for the G should be balanced. So unlike the training on MNIST, I reduce the beta to 0.1 because sometimes the loss_imitate can not decrease, and the G can just minimize the loss_diversity.
> 
> I have tried adding the random seed, but I don't know why it didn't work. So maybe you should try several times to train the substitute model. You can observe the losses. If the training is collapsed, you stop it early. The good training for DaST-P should be like the second example in the cifar10 log file. But for DaST-L, The training will always be collapsed after multiple epochs, you can observe it in my paper, the performance of DaST-L is not good for CIFAR-10.

The new updated code `dast_cifar10.py` works for me on the cifar-10 experiments. Sincerely thanks!
